### PR TITLE
LPS-165672 Add autom. test | FDSCustomView#HasActionsOnlyOnEllipsis

### PR DESCRIPTION
**Pull request by mateussandes23**
Background
Create automation tests for FDSCustomView

**Test Plan:**

- Given FDS Sample Portlet
- And Given Custom tab
- And Given created custom view
- When open ellipsis next to custom view
- Then actions to save and delete views are available in dropdown menu
- When open select custom view button // Default View button next to ellipsis
- Then actions to save and delete views are not available in dropdown menu

JIRA Ticket:
https://issues.liferay.com/browse/LPS-168707

### CODE REVIEW

**Comment by mateussandes23** [comment]
ci:test:sf

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase [hunk 1]
```diff
@@ -239,6 +239,46 @@ definition {
...
+			Pause(locator1 = "5000");
+
+			AssertElementNotPresent(
+				key_itemName = "Save View",
+				locator1 = "FrontendDataSet#SELECT_OPTION_CUSTOM_VIEW");
+
+			AssertElementNotPresent(
+				key_itemName = "Delete View",
+				locator1 = "FrontendDataSet#SELECT_OPTION_CUSTOM_VIEW");
+		}
```
Comment (mateusralv) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase:279]: Add "Given, when then" in the task Title

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase [hunk 2]
```diff
@@ -239,6 +239,46 @@ definition {
...
+
+		task ("open select custom view button") {
+			FDSCustomView.changeCustomView(key_itemName = "Default View");
+
+			Click(locator1 = "FrontendDataSet#SELECT_OPTION_HEADER");
+		}
+
+		task ("actions to save and delete views are not available in dropdown menu") {
+			Pause(locator1 = "5000");
```
Comment (mateusralv) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase:270]: could you remove the pause? we put just to view the results 😄

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path [hunk 3]
```diff
@@ -211,14 +211,29 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>INPUT_CUSTOM_VIEW_NAME2</td>
+	<td>//input[contains(@id,'LabelInput')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>SAVE_BUTTON</td>
+	<td>//button[@class='dropdown-item'][normalize-space(text())='Save View']</td>
```
Comment (mateusralv) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path:236]: could you use `SELECT_OPTION_CUSTOM_VIEW` with `key_itemName = 'Save View'`? and delete this path

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path [hunk 4]
```diff
@@ -232,8 +247,13 @@
 	<td></td>
 </tr>
 <tr>
-	<td>HEADER_ROW</td>
-	<td>xpath=(//div[contains(@class, "dnd-tr")])[1]</td>
+	<td>DELETE_VIEW_BUTTON</td>
+	<td>//button[@class='dropdown-item'][normalize-space(text())='Delete View']</td>
+	<td></td>
+</tr>
```
Comment (mateusralv) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path:253]: same here 😄

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase [hunk 5]
```diff
@@ -239,6 +239,46 @@ definition {
...
+		}
+
+		task ("open ellipsis next to custom view") {
+			Click(locator1 = "FrontendDataSet#SELECT_OPTION_HEADER");
+		}
+
+		task ("actions to save and delete views are available in dropdown menu") {
+			AssertElementPresent(
+				key_itemName = "Save View",
+				locator1 = "FrontendDataSet#SELECT_OPTION_CUSTOM_VIEW");
```
Comment (mateusralv) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase:256]: could you create the macro and use it inside the test? Thank you 
```
macro assertMenuItem {
		AssertElementPresent(
				key_itemName = "${key_menuItem}",
				locator1 = "FrontendDataSet#SELECT_OPTION_CUSTOM_VIEW");
	}
```

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path [hunk 6]
```diff
@@ -232,8 +247,13 @@
...
+<tr>
+	<td>DELETE_BUTTON</td>
+	<td>//button[@class='btn btn-primary']</td>
```
Comment (mateusralv) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path:256]: That path is not being used in the test 😄

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path [hunk 7]
```diff
@@ -211,14 +211,29 @@
 
 <!--CUSTOM VIEWS-->
 
+<tr>
+	<td>OPEN_SELECT_FIELD</td>
+	<td>//button[contains(@aria-label, "Open Fields Menu")]</td>
+	<td></td>
+</tr>
```
Comment (mateusralv) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path:218]: That path is not being used in the test 😄

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path [hunk 8]
```diff
@@ -211,14 +211,29 @@
...
 <tr>
 	<td>SELECT_OPTION_CUSTOM_VIEW</td>
-	<td>//button[contains(@class,'dropdown-item') and contains(text(), '${key_itemName}')]</td>
+	<td>//button[@class='dropdown-item'][normalize-space(text())='${key_itemName}']</td>
+	<td></td>
+</tr>
+<tr>
+	<td>INPUT_CUSTOM_VIEW_NAME2</td>
+	<td>//input[contains(@id,'LabelInput')]</td>
+	<td></td>
```
Comment (mateusralv) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path:232]: That path is not being used in the test 😄

**Review by mateusralv — CHANGES_REQUESTED** [review]

**Comment by mateusralv** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path (hunk 6 above)
Comment (mateussandes23) [reply portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path:256]: Hey @mateusralv, this path is being used on `deleteView` macro .

**Pull request closed without merging**